### DISCUSSION TITLE
Consider switching the default branch to main? (Would dependencies prevent this change?)

### DISCUSSION THREAD

**Discussion by annegentle** [daattali/beautiful-jekyll#1008] [General]
I looked in the Issues and Discussions but hadn't seen anyone bring this up yet - and I'm not sure if there's any appetite for this change. 

Since this is a great site for learning more about Git, GitHub, and Jekyll, have you considered switching from `master` as the default branch to `main` now that all new repositories use `main` as the default branch? I sense that more newcomers to GitHub will look for `main` rather than `master`. 

The GitHub Pages product was changed to enable any branch name about two years ago: 
https://github.blog/changelog/2020-07-31-build-and-deploy-github-pages-from-any-branch-beta/
And for GitHub Pages, the docs screenshots now show `main` for the publishing source:
https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#choosing-a-publishing-source

I searched the repo and see that the `staticman.yml` uses `master` in the default file, and other entries such as `_config.yml` are commented out since that entry is pointed to the optional use of Staticman. Completely understand if there's some outside dependency that would prevent this branch name change. I wanted to bring it up here to see if you and the community would want to see this change. It's not really a PR thing (other than explanations in the config and README files) since it's settings on the repo itself. I'd be happy to help with explanations in a PR as needed. (Promise, they won't be this long-winded. 😆)

Thanks,
Anne

**Comment by daattali** [comment]
Thanks for opening this up. I would be happy to change it to `main` if I was guaranteed that it would not affect any of the integrated features and any users. I'm spending time recently making major updates to other packages I own so I don't have time for thorough testing on beautifuljekyll for something like this. I'll gladly receive help in testing all the 3rd party integrations (in config.yml) and updating any necessary documentation/code to make this change.

**Comment by annegentle** [comment]
Got it, thanks for the quick turnaround on the reply. 

To get the scope correct, is this a comprehensive list of 3rd party integrations? 

Google Analytics gtag
Google Analytics (original ID)
Google Tag Manager ID
Cloudflare Analytics
Matomo Web stats (Piwik)
Disqus
Facebook Comments 
Facebook Apps
CommentBox
Utterances
Staticman
giscus

**Reply by daattali** [reply]
Yes that seems like a complete list
